pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -201,7 +201,8 @@ func (p *staticPolicy) validateState(s state.State) error {
 		}
 		// state is empty initialize
 		allCPUs := p.topology.CPUDetails.CPUs()
-		s.SetDefaultCPUSet(allCPUs)
+		nonReservedCPUs := allCPUs.Difference(p.reservedCPUs)
+		s.SetDefaultCPUSet(nonReservedCPUs)
```

Comment:
thanks for the fix!
is the recovery flow (reading back state from cpu_manager_state) handled correctly? (see also lines 208-215 in policy_static.go)
Let's check (add tests) to side functions like GetAvailableCPUs GetAllocatableCPUs to check they return the expected values after this change

Reply:
@ffromani Do you think we should make this configurable? Just to allow the old behavior for specific use cases. I just learned about another use case where this change is actually wanted, however I am worried the change will affect some of our users. Especially if this is ever backported.

The change here is small enough to allow for an `if`.

WDYT?

Reply:
I think I changed my mind. The old behavior is probably possible via a simple runtime hook and it would just pollute kubelet.